test/consumer.ts

```diff
@@ -729,6 +775,42 @@ describe('Consumer', () => {
       sandbox.assert.callCount(handleMessage, 0);
     });
 
+    it('respects semaphore capacity', async () => {
+      const expectedAcquireCount = 16;
+      const expectedHandleMessageCount = 15;
+      const expectedReleaseCount = 10;
+      const expectesNotReadCount = 118;
+      const semaphoreCapacity = 5;
+      let handleMessageCount = 0;
+      const testSem = new TestSemaphore(semaphoreCapacity);
+
+      consumer = new Consumer({
+        queueUrl: 'some-queue-url',
+        messageAttributeNames: ['attribute-1', 'attribute-2'],
+        region: 'some-region',
+        handleMessage: () => new Promise((resolve) => {
+          handleMessageCount += 1;
+          setTimeout(resolve, 200);
+        }),
+        semaphore: testSem,
+        sqs
+      });
+
+      consumer.start();
+
+      for (let i = 0; i < 40; i += 1) {
+        await clock.tickAsync(15);
+      }
+
+      consumer.stop();
+
+      assert.strictEqual(testSem.count, semaphoreCapacity, 'Expected semaphore count to be equal capacity');
+      assert.strictEqual(testSem.acquireCount, expectedAcquireCount, 'Expected acquire to be called ' + expectedAcquireCount + ' times');
+      assert.strictEqual(testSem.releaseCount, expectedReleaseCount, 'Expected release to be called ' + expectedReleaseCount + ' times');
+      assert.strictEqual(handleMessageCount, expectedHandleMessageCount, 'Expected handle message to be called ' + expectedHandleMessageCount + ' times');
+      assert.strictEqual(testSem.notReadyCount, expectesNotReadCount, 'Expected semaphore to be not read' + expectesNotReadCount + ' times');
```

Comment:
Seguindo a nossa convenção de escrita de assertions:
```ts
expect(testSem.count).to.equal(semaphoreCapacity);
```

Acho que não precisa testar a quantidade de acquires, releases, e not-ready's. O importante é só garantir que o `testSem.count` fica estacionado em `semaphoreCapacity`.